test/src/common/bridgePowerController_ut.cpp

```diff
@@ -28,6 +28,18 @@ BaseType_t (*custom_fakes[])(RTCTimeAndDate_t *timeAndDate) = {
     rtcGet_custom_fake,
 };
 
+static uint8_t persistent_fake_pin_val;
+bool io_read_custom_fake(const void *_, uint8_t *pinval) {
+  (void)_;
+  *pinval = persistent_fake_pin_val;
+  return true;
+}
+bool io_write_custom_fake(const void *_, uint8_t pinval) {
+  (void)_;
+  persistent_fake_pin_val = pinval;
+  return true;
+}
+
```

Comment:
Without these custom fakes to track the persistent pin state, the "golden path" test below was actually just always succeeding! It was completely wrong and not helping us detect regressions. That test definitely should've started failing at least when we removed the bus power cycle after the init period, but it didn't. I don't know why it started failing in CI while I worked on this PR, but I'm glad it did! @russelldeguzman @victorsowa12 I'd like to verbally walk through these tests in a review to make sure we all understand them.